sway/commands/bind.c

```diff
@@ -69,22 +69,24 @@ bool binding_key_compare(struct sway_binding *binding_a,
 	return true;
 }
 
-struct cmd_results *cmd_bindsym(int argc, char **argv) {
+static struct cmd_results * cmd_bindsym_or_bindcode(int argc, char **argv, bool bindcode) {
+	const char* bindtype = bindcode ? "bindcode" : "bindsym";
+
 	struct cmd_results *error = NULL;
-	if ((error = checkarg(argc, "bindsym", EXPECTED_MORE_THAN, 1))) {
+	if ((error = checkarg(argc, bindtype, EXPECTED_MORE_THAN, 1))) {
 		return error;
 	}
 
 	struct sway_binding *binding = calloc(1, sizeof(struct sway_binding));
 	if (!binding) {
-		return cmd_results_new(CMD_FAILURE, "bindsym",
+		return cmd_results_new(CMD_FAILURE, bindtype,
```

Comment:
Side note, you can also get this from `argv[-1]`

Reply:
I'd prefer not to change this as it isn't immediately clear that `argv` has a valid value at `-1`; at least, no other command uses this trick. Also, the function changes the values of `argc` and `argv`, so this would only change the line which sets `bindtype`, making it more difficult to see the possible values.